flang/lib/Semantics/expression.cpp

```diff
@@ -2546,15 +2557,33 @@ static int GetMatchingDistance(const characteristics::DummyArgument &dummy,
       return 2;
     }
   } else if (*dummyDataAttr == common::CUDADataAttr::Managed) {
-    if (!actualDataAttr || *actualDataAttr == common::CUDADataAttr::Device) {
+    if (!actualDataAttr) {
+      if (isGpuUnified) {
```

Comment:
Would be more readable as `return isGpuUnified ? 1 : isGpuManaged ? 0 : cudaInfMatchingValue;`.

Reply:
Yeah I like it more like that. Updated